scipy/stats/stats.py

```diff
@@ -558,11 +558,12 @@ def tmean(a, limits=None, inclusive=(True, True), axis=None):
 
     """
     a = asarray(a)
+    a = a.astype(float)
```

Comment:
You can potentially avoid a copy here by passing `copy=false` to `astype`, or even better:
```suggestion
    a = asarray(a, dtype=np.float64)
```